File: backend/models/conversation.py
from pydantic import BaseModel
from typing import Optional, List, Literal


class ConversationTurn(BaseModel):
    role: Literal["user", "assistant", "tool"]
    content: str
    timestamp: Optional[str] = None
    tokens: Optional[int] = None


class ConversationResponse(BaseModel):
    id: str
    title: str
    date: str
    tokens: int = 0
    turns: int = 0
    project: Optional[str] = None
    project_folder: Optional[str] = None


class ConversationDetail(ConversationResponse):
    messages: List[ConversationTurn] = []

File: frontend/src/types/conversation.ts
export interface Conversation {
  id: string;
  title: string;
  date: string;
  tokens: number;
  turns: number;
  project?: string;
}

export interface ConversationResponse {
  id: string;
  title: string;
  date: string;
  tokens: number;
  turns: number;
  project?: string;
  project_folder?: string;
}

export interface ConversationTurn {
  role: 'user' | 'assistant' | 'tool';
  content: string;
  timestamp?: string;
  tokens?: number;
}

export interface ConversationDetail {
  id: string;
  title: string;
  date: string;
  tokens: number;
  turns: number;
  messages: ConversationTurn[];
}